csrc/class_geometry.cpp

```diff
@@ -8,11 +8,27 @@ JNIEXPORT jint JNICALL Java_mapnik_Geometry_getType
   (JNIEnv *env, jobject gobj)
 {
 	PREAMBLE;
-	mapnik::geometry_type* g=LOAD_GEOMETRY_POINTER(gobj);
-	return g->type();
+	mapnik::geometry::geometry<double>* g=LOAD_GEOMETRY_POINTER(gobj);
+	return mapnik::geometry::geometry_type(*g);
 	TRAILER(0);
 }
 
+
+struct VertexCollector
```

Comment:
What I'm trying to do here is to collect all the vertices in an arbitrary `geometry<double>` into an `std::vector`, using `geometry:: vertex_processor`. I *think* it should work, but it is certainly inefficient to do this on every call to `getNumPoints` or `getVertex`. => The (internal) Java interface should be refactored to do this more efficiently.